### PR TITLE
ci: rework build step

### PR DESCRIPTION
**Pull request by phbelitz**
Removed the signing of the build image and creation of SBOM to reduce complexity. Also moved the code of the context and build action into the reusable build workflow.

## Checklist

- [x] PR is rebased to/aimed at branch `develop`
- [x] PR follows [Contributing Guide](https://github.com/sse-secure-systems/connaisseur/blob/master/docs/CONTRIBUTING.md)
- [x] Added tests (if necessary)
- [x] Extended README/Documentation (if necessary)
- [x] Adjusted versions of image and Helm chart in `Chart.yaml` (if necessary)

### CODE REVIEW

File: .github/workflows/.reusable-build.yml [hunk 1]
```diff
@@ -79,21 +141,46 @@ jobs:
...
-          repo_token: ${{ secrets.GITHUB_TOKEN }}
-          cosign_private_key: ${{ secrets.COSIGN_PRIVATE_KEY }}
-          cosign_password: ${{ secrets.COSIGN_PASSWORD }}
+          push: true
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+          file: build/Dockerfile
+          labels: ${{ needs.context.outputs.build_labels }}
+          tags: ${{ steps.tags.outputs.tags }}
+          sbom: false  # Duplicates SBOMs manually created below
```
Comment (Starkteetje) [inline .github/workflows/.reusable-build.yml:173]: this comment is no longer accurate

File: test/integration/common_functions.sh [hunk 2]
```diff
@@ -180,20 +180,12 @@ ghcr_update() { # $1: file
     yq e -i '.kubernetes.deployment.image.tag = env(TAG)' $1
     # add imagePullSecrets
     yq e -i '.kubernetes.deployment.image.imagePullSecrets[0].name = env(IMAGEPULLSECRET)' $1
-    # add cosign validator:
```
Comment (Starkteetje) [inline test/integration/common_functions.sh:183]: This change completely changes the context of this function. It should at least be renamed. Probably, we can get rid of it, too

Comment (Starkteetje) [reply test/integration/common_functions.sh:183]: Maybe even the whole stuff related to GHCR can be scrubbed from the testcases

Comment (phbelitz) [reply test/integration/common_functions.sh:183]: the ghcr image is still present, this much hasn't changed. only thing that has changed is that the image is no longer signed. thus the function is still valid.

**Review by Starkteetje — APPROVED** [review]

**Pull request merged**